plugins/search-backend-node/src/Scheduler.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from 'winston';
+
+type TaskEnvelope = {
+  task: Function;
+  interval: number;
+};
+
+/**
+ * TODO: coordination, error handling
+ */
+
+export class Scheduler {
```

Comment:
Any thoughts on breaking things out of the scaffolder to re-use some of the task queue logic we got there? Or do you think this'll be simple enough to not be worth it?

There's also our ongoing catalog work that might have some similarities.

Reply:
An eventual convergence may make sense (would be curious to hear what, if any, similar work is going on in relation to coordination of scaffolder tasks / catalog refreshing across horizontally scaled processes.

But probably fine to punt on it to a later stage while we're here, wiring up an in-process search engine.

Reply:
Yes I would like to sync on you all both on ongoing work around similar things for the scaffolder and the catalog! And if our use cases are similar enough we can think about move towards a shared solution as a next step!